django_toolkit/concurrent/locks.py

```diff
@@ -0,0 +1,59 @@
+from django.core.cache import caches
+
+
+class LockActiveError(Exception):
+    pass
+
+
+class Lock(object):
+
+    def __init__(self):
+        self.active = False
+
+
+class LocalMemoryLock(Lock):
+    """
+    A context manager to handle a lock status in memory
+
+    The active status is True on __enter__ and False on __exit__
+    """
+
+    def __enter__(self):
+
+        if self.active:
+            raise LockActiveError('Lock is already active')
+
+        self.active = True
+        return self
+
+    def __exit__(self, *args, **kwargs):
+        self.active = False
+
+
+class CacheLock(Lock):
```

Comment:
What about some docstring here? You could include `__init__` parameter descriptions, also.